rviz_default_plugins/src/rviz_default_plugins/displays/pointcloud/point_cloud_common.cpp

```diff
@@ -267,7 +267,9 @@ void PointCloudCommon::updateBillboardSize()
   }
   for (auto & cloud_info : cloud_infos_) {
     cloud_info->cloud_->setDimensions(size, size, size);
-    cloud_info->selection_handler_->setBoxSize(getSelectionBoxSize());
+    if (cloud_info->selection_handler_) {
+      cloud_info->selection_handler_->setBoxSize(getSelectionBoxSize());
+    }
```

Comment:
Hm.  While I definitely see that this is a problem, I'm kind of concerned about the integrity of the selection_handler in general.  My questions/concerns are:

1. Why are we allocating/deallocating the handler at all when we set it not selectable?  It seems simpler to me to just allocate it in the constructor and never touch it again until we destroy the class.
2. There is another place that messages with the selection_handler pointer, `onDisable`.  It kind of feels to me that that should call `setSelectable(false)` so we concentrate the logic all in one place, but I'm not sure.
3. Are we sure that accessing the selection_handler is always done on the same thread?  Otherwise, it looks pretty racy to access it like this (and this sort of gets back to my first point as well).

Reply:
Yes, good points. I'll have a closer look to get some answers.